make init 실패

- macOS 버전: <!-- 숫자로 써주세요. 시스템 메뉴에서  -> 이 맥에 관하여 --> 모하비 10.14.2
- 구름 버전: <!-- 시스템 메뉴의 입력기 아이콘에서 ☁️ -> 구름 입력기에 관하여 -->
- 사용중인 한글/영문 배열:
- 문제가 발생하는 애플리케이션 이름과 버전:
- 문제가 있는 동작:
- 기대하는 정상 동작:
- 애플 기본 한글 입력기의 동작:
- 재현 방법: <!-- 누구든 따라할 수 있도록 애플리케이션 시작 단계부터 상세히 적어주세요 -->
make init 시 아래와 같은 에러 발생
```
brew install shellcheck swiftformat
Warning: shellcheck 0.7.0 is already installed and up-to-date
To reinstall 0.7.0, run `brew reinstall shellcheck`
Warning: swiftformat 0.40.12 is already installed and up-to-date
To reinstall 0.40.12, run `brew reinstall swiftformat`
gem install cocoapods xcpretty
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.3.0 directory.
gem-install failed. Try sudo make gem-install.
make: *** [gem-install] Error 1
```
<!--
버그 신고의 경우 위 서식을 가능한 채워서 제출해 주시면 큰 도움이 됩니다.
채울 수 없는 칸은 비워 두셔도 됩니다.

유료 애플리케이션 및 Cocoa가 아닌 앱의 호환성 문제는 빠른 시일 내에 해결되지 않을 수 있습니다.
내장 한글 입력기와 구름 입력기에서의 동작을 비교 확인해 주세요.
두 입력기에서 동시에 발생하는 같은 오류는 입력기가 아닌 애플리케이션의 오류일 가능성이 높습니다.

아래에 예시가 포함되어 있으니 참고하세요.

예시:

- macOS 버전: 10.15.1
- 구름 버전: 1.20.8
- 사용중인 한글/영문 배열: 세벌식 최종 순아래 / 쿼티
- 문제가 발생하는 애플리케이션 이름과 버전: iTerm2 3.2.9
- 문제가 있는 동작: 한글 낱자를 지울 때 마지막 자소를 지우려 하면 백스페이스를 2번 눌러야 합니다.
- 기대하는 정상 동작: 다른 애플리케이션과 같이 마지막 자소도 백스페이스 1번에 지워져야 합니다.
- 애플 기본 한글 입력기의 동작: 마지막 자소도 백스페이스 1번에 지워집니다.
- 재현 방법: <BS>를 백스페이스라고 할때,
 - iTerm 실행
 - 한 입력  -> 한
 - <BS><BS> 입력 -> ㅎ
 - <BS> 입력 -> ㅎ
마지막 단계에서 ㅎ이 남지 않고 삭제되어야 합니다.
-->

--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ format:
 	swiftformat OSXCore/ OSX/ GureumTests/ Preferences/ OSXTestApp/
 
 brew-install:
-	@command -v brew >/dev/null || { echo 'Homebrew is not installed. See https://brew.sh'; exit 1; }
+	@command -v brew >/dev/null || { echo 'Error: Homebrew is not installed. See https://brew.sh'; exit 1; }
 	@if ! command -v shellcheck >/dev/null; then \
 		echo 'brew install shellcheck'; \
 		brew install shellcheck; \
@@ -26,11 +26,11 @@ brew-install:
 gem-install:
 	@if ! command -v pod >/dev/null; then \
 		echo 'gem install cocoapods'; \
-		gem install cocoapods || { echo 'gem-install failed. Try sudo make gem-install.'; exit 1; }; \
+		gem install cocoapods || { echo 'Error: gem-install failed. Try sudo make gem-install.'; exit 1; }; \
 	fi
 	@if ! command -v xcpretty >/dev/null; then \
 		echo 'gem install xcpretty'; \
-		gem install xcpretty || { echo 'gem-install failed. Try sudo make gem-install.'; exit 1; }; \
+		gem install xcpretty || { echo 'Error: gem-install failed. Try sudo make gem-install.'; exit 1; }; \
 	fi
 
 .PHONY: all init format brew-install gem-install

--- a/Makefile
+++ b/Makefile
@@ -14,10 +14,23 @@ format:
 
 brew-install:
 	@command -v brew >/dev/null || { echo 'Homebrew is not installed. See https://brew.sh'; exit 1; }
-	brew install shellcheck swiftformat
+	@if ! command -v shellcheck >/dev/null; then \
+		echo 'brew install shellcheck'; \
+		brew install shellcheck; \
+	fi
+	@if ! command -v swiftformat >/dev/null; then \
+		echo 'brew install swiftformat'; \
+		brew install swiftformat; \
+	fi
 
 gem-install:
-	@echo 'gem install cocoapods xcpretty'
-	@gem install cocoapods xcpretty || { echo 'gem-install failed. Try sudo make gem-install.'; exit 1; }
+	@if ! command -v pod >/dev/null; then \
+		echo 'gem install cocoapods'; \
+		gem install cocoapods || { echo 'gem-install failed. Try sudo make gem-install.'; exit 1; }; \
+	fi
+	@if ! command -v xcpretty >/dev/null; then \
+		echo 'gem install xcpretty'; \
+		gem install xcpretty || { echo 'gem-install failed. Try sudo make gem-install.'; exit 1; }; \
+	fi
 
 .PHONY: all init format brew-install gem-install

